packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-dependencies.js

```diff
@@ -0,0 +1,155 @@
+import { basename, dirname, extname, join } from "path";
+import { copy, readJson, writeJson } from "fs-extra";
+import { promises } from "fs";
+import { rm } from "shelljs";
+
+export function collectDependencies(
+    externals,
+    outputDir,
+    shouldCopyNodeModules = true,
+    widgetName,
+    shouldRemoveNodeModules = true,
+    isJSAction = false
```

Comment:
Are you sure we must have this config? Overall having flags that majorly modify a behavior is not a good practice. I also don't see a big difference between how dependencies are handled for widgets vs actions.

Reply:
Yes, it actually might be better to have a separate plugin for jsActions that would copy of all their dependencies. Because iiuc this is what you are trying to achieve - for js actions copy all dependencies diregarding of whether they are native or not.

Reply:
Alternatively you can split this plugin into two: one for generating json file and one for copying stuff. Then maybe the one for copying can be configured to copy only native or all dependencies (you might try to express it with `filter` function parameter e.g.)

Reply:
Maybe at least name it more generically, like `copyOnlyNativeDependencies`

Reply:
Its not only dependency, it is collecting dependencies for JSActions and also Native deps for JS and Widgets